server/src/test/java/io/crate/planner/operators/JoinTest.java

```diff
@@ -99,14 +99,14 @@ private LogicalPlan createLogicalPlan(QueriedSelectRelation mss, TableStats tabl
             () -> clusterService.state().nodes().getMinNodeVersion()
         );
         SubqueryPlanner subqueryPlanner = new SubqueryPlanner((s) -> logicalPlanner.planSubSelect(s, plannerCtx));
-        return JoinPlanBuilder.buildJoinTree(
+        var plan = JoinPlanBuilder.buildJoinTree(
             mss.from(),
             mss.where(),
             mss.joinPairs(),
             new SubQueries(Map.of(), Map.of()),
-            rel -> logicalPlanner.plan(rel, plannerCtx, subqueryPlanner, true),
-            txnCtx.sessionSettings().hashJoinsEnabled()
+            rel -> logicalPlanner.plan(rel, plannerCtx, subqueryPlanner, true)
         );
+        return logicalPlanner.optimizer.optimize(plan, tableStats, txnCtx);
```

Comment:
Does it maybe make sense to keep this as it was, and add a second step of optimization and assertions?

Reply:
It's fine it was more of a testing improvement question, if it would be better to separate the test of nl changes from the (now) optimization to convert them to hash joins.

We can re-think this later on when we introduce new optimizations.